src/models/disassemblyoutput.cpp

```diff
@@ -25,34 +25,52 @@
   along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-#include "data.h"
 #include "disassemblyoutput.h"
-#include <QStandardPaths>
+#include "data.h"
 #include <QApplication>
-#include <QProcess>
 #include <QDebug>
+#include <QProcess>
+#include <QRegularExpression>
+#include <QStandardPaths>
 
 static QVector<DisassemblyOutput::DisassemblyLine> objdumpParse(const QByteArray &output)
 {
     QVector<DisassemblyOutput::DisassemblyLine> disassemblyLines;
 
     QTextStream stream(output);
     QString asmLine;
+    const QRegularExpression disassemblyRegex(QStringLiteral("^   [0-9a-f]{4,}:       [0-9a-f ]{2,}"));
     while (stream.readLineInto(&asmLine))
     {
         if (asmLine.isEmpty() || asmLine.startsWith(QLatin1String("Disassembly")))
             continue;
 
-        const auto asmTokens = asmLine.splitRef(QLatin1Char(':'));
-        const auto addrLine = asmTokens.value(0).trimmed();
+        const int colonIndex = asmLine.indexOf(QLatin1Char(':'));
+        const int angleBracketIndex = asmLine.indexOf(QLatin1Char('<'));
 
-        bool ok = false;
-        const auto addr = addrLine.toULongLong(&ok, 16);
-        if (!ok) {
-            qWarning() << "unhandled asm line format:" << addrLine;
+        // skip function header line as it is not interesting for us
+        if (angleBracketIndex > 0 && colonIndex > angleBracketIndex) {
             continue;
         }
 
+        // we don't care about the file name
+        if (asmLine.startsWith(QLatin1Char('/')) && asmLine.contains(QStringLiteral("file format"))) {
+            continue;
+        }
+
+        quint64 addr = 0;
+        if (disassemblyRegex.match(asmLine).hasMatch()) {
+            const auto asmTokens = asmLine.splitRef(QLatin1Char(':'));
+            const auto addrLine = asmTokens.value(0).trimmed();
```

Comment:
considering you are already doing an expensive regex match above, please at the very least leverage it properly!

add a capture group for the address part and then use `match.capturedRef(1).toULongLong`